command/agent/auth/auth.go

```diff
@@ -104,7 +112,15 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 		return errors.New("auth handler: nil auth method")
 	}
 
-	backoff := newAgentBackoff(ah.maxBackoff)
+	if ah.minBackoff <= 0 {
+		ah.minBackoff = defaultInitialBackoff
+	}
+
+	backoff := newAgentBackoff(ah.minBackoff, ah.maxBackoff)
+
+	if backoff.current >= backoff.max {
```

Comment:
If we add `initial`, the check could be against that. We could also cap the max to initial instead of returning an error, but I'll leave it up to you on how we what to handle that.